### DISCUSSION TITLE
RRF Integration

### DISCUSSION THREAD

**Discussion by Superbrain8** [Donkie/Spoolman#282] [General]
Ive been following the project for a little now.
used to run it with Moonraker in the past, but due to some issues with Klipper overall ive decided to move now Entirely to RRF.

To keep the ability to continou using this project ive wrote a small app in JS that allows the use with RRF.

Still have to write some documentation for it and the integration into RRF is no where as tight as with Moonraker.
but it works. it keeps track of Filament usage.
About how precise it is i still have to do some testing.

**Comment by P-C-R** [comment]
@Superbrain8  can you pint me to the code of your RRF integration

**Reply by Superbrain8** [reply]
https://github.com/Superbrain8/spoolman_rrf_integration
it has a few issues i where not able to get around yet.
also it doesnt work as reliable as i would like to.

overall it works by querying the machine state model of RRF and checking for the current extruder position every few seconds.
the current position gets stored in a value until the next loop where it fetches data again and then subtracts the new position from the old Position to get the filament used. that data gets then send to Spoolman.
For identifying the used spool i use some hacky way with the Filament Manager in RRF
